### ISSUE TITLE
Element#click_no_wait does not click elements in a frame

### ISSUE THREAD

**Issue by jkotests** [watir/watir-classic#45]
When trying to set a file field that is located within a frame, the "Choose File to Upload" dialog never opens, causing the script to timeout.

This issue does not occur if the file field is not within a frame.

**Steps to Reproduce - Using w3schools:**
1. Run the following script (using any existing file for upload):

``` ruby
require 'watir-classic'
browser = Watir::Browser.new
browser.goto('http://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_input_type_file')
browser.frame(:index, 1).file_field.set('c:\users\jko\desktop\file.txt')
```

--> Notice that the "Choose File to Upload" dialog never opens and that the following exception occurs.

``` ruby
RAutomation::UnknownWindowException: Window with locator {:index=>0, :title=>/^choose file( to upload)?|Elegir archivos para cargar$/i} doesn't exist or is not visible!
        from C:/Ruby193/lib/ruby/gems/1.9.1/gems/rautomation-0.7.3/lib/rautomation/window.rb:226:in `rescue in wait_until_present'
        from C:/Ruby193/lib/ruby/gems/1.9.1/gems/rautomation-0.7.3/lib/rautomation/window.rb:224:in `wait_until_present'
        from C:/Ruby193/lib/ruby/gems/1.9.1/gems/rautomation-0.7.3/lib/rautomation/window.rb:214:in `text_field'
        from C:/Ruby193/lib/ruby/gems/1.9.1/gems/watir-classic-3.3.0/lib/watir-classic/dialogs/file_field.rb:19:in `set_file_name'
        from C:/Ruby193/lib/ruby/gems/1.9.1/gems/watir-classic-3.3.0/lib/watir-classic/dialogs/file_field.rb:7:in `set'
        from (irb):10
        from C:/Ruby193/bin/irb:12:in `<main>'

```

**Steps to Reproduce - Locally:**
1. Create a file called "file_field.htm", which contains a file field:

``` html
<html>
    <body>
        <input type="file" id="myfile" />
    </body>
</html>
```
1. Create a file called "main.htm", which includes the above file in a frame:

``` html
<html>
    <body>
        <iframe src="file_field.htm">
    </body>
</html>
```
1. Set the file field:

``` ruby
require 'watir-classic'
browser = Watir::Browser.new
browser.goto('C:\Users\jko\Desktop\main.htm')
browser.frame.file_field.set 'c:\users\jko\desktop\file.txt'
```

--> Notice that the RAutomation::UnknownWindowException mentioned above occurs.

**Environment:**
- OS: Windows Vista
- Browser: IE9
- Ruby: 1.9.3p327
- Gems: watir (4.0.2), watir-classic (3.3.0), rautomation (0.7.3)

**Comment by slvganesh** [comment]
I have a problem where the same click_no_wait is working in my local machine but not in server environment and that too with 2 or more IE instance. With single instance of IE, click_no_wait is working in server environments without any issue. Any help?

**Comment by jarmo** [comment]
I've written a blog post about debugging `#click_no_wait` problems here http://itreallymatters.net/post/378669758/debugging-watirs-click-no-wait-method-problems - it is definitely quite an old one, but main logic is still the same when it comes to debugging that kind of problems.

**Comment by slvganesh** [comment]
Thanks Jarmo, I've narrowed down my problem a bit with your help. Now I have ended up with the below issue: 
rescue in attach_browser_window': Unable to locate a window with hwnd of 853530 (Watir::Exception::NoMatchingWindowFoundException)

I noticed that the hwnd seems to be correct (maps to the intended IE instance). This issue happens when I have two IE browsers open. Even Browser.attach :title, /title/ only works with single IE browser window but not with two. 

It would be helpful if I get some hints to debug this issue.

**Comment by slvganesh** [comment]
Thanks Jarmo, I've narrowed down my problem a bit with your help. Now I have ended up with the below issue: 
rescue in attach_browser_window': Unable to locate a window with hwnd of 853530 (Watir::Exception::NoMatchingWindowFoundException)

I noticed that the hwnd seems to be correct (maps to the intended IE instance). This issue happens when I have two IE browsers open. Even Browser.attach :title, /title/ only works with single IE browser window but not with two. 

It would be helpful if I get some hints to debug this issue.

**Comment by slvganesh** [comment]
Thanks Jarmo, I've narrowed down my problem a bit with your help. Now I have ended up with the below issue: 
rescue in attach_browser_window': Unable to locate a window with hwnd of 853530 (Watir::Exception::NoMatchingWindowFoundException)

I noticed that the hwnd seems to be correct (maps to the intended IE instance). This issue happens when I have two IE browsers open. Even Browser.attach :title, /title/ only works with single IE browser window but not with two. 

It would be helpful if I get some hints to debug this issue.
